### PR TITLE
Set up package with typescript

### PR DESCRIPTION
**Pull request by MBLHarrison**
**Create typescript environment**
Project now includes
+ typscript
+ jest
+ tslint
+ prettier

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,5 +33,42 @@
...
+  "lint-staged": {
+    "*.html": [
+      "prettier --write",
+      "git add"
+    ],
+    "*.ts": [
+      "eslint --fix",
+      "git add"
+    ],
+    "*.scss": [
```
Comment (huwmartin) [inline package.json:69]: Probably don't need this

File: package.json [hunk 2]
```diff
@@ -21,5 +33,42 @@
...
+    "lib": "lib"
+  },
+  "dependencies": {},
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "*.html": [
```
Comment (huwmartin) [inline package.json:61]: Probably don't need this

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
+  "env": {
+    "es6": true,
+    "node": true
+  },
+  "extends": ["eslint:recommended", "plugin:@typescript-eslint/eslint-recommended"],
```
Comment (huwmartin) [inline .eslintrc.json:6]: Do you want to add prettier recommended like you had for tslint here

File: package.json (hunk 2 above)
Comment (MBLHarrison) [reply package.json]: cheers

File: .eslintrc.json (hunk 3 above)
Comment (MBLHarrison) [reply .eslintrc.json]: cheers

File: package.json (hunk 1 above)
Comment (MBLHarrison) [reply package.json]: cheers

**Pull request merged**